### ISSUE TITLE
Question about the evaluation

### ISSUE THREAD

**Issue by 07hyx06** [barbararoessle/ganerf#3]
Hi, thanks for the impressive work! I have a question on why GANeRF can obtain better view consistency than Nerfacto + ControlNet.

In my understanding, both GANeRF and Nerfacto + ControlNet use a pure 2D refinement network to improve the results. When training the generator in GANeRF, the gradient does not propagate to the NeRF network; the final results of GANeRF are also produced from the 2D generator, not the NeRF, just like the Nerfacto + ControlNet baseline. So GANeRF also can not guarantee strict view consistency as the results are rendered from a 2D network, not a NeRF.

I think my question may stem from my misunderstanding of the implementation of the baseline variant Ours w/o discriminator. There are two discriminators in GANeRF. Does Ours w/o discriminator mean removing the discriminator in the "NeRF Optimization with Discriminator" stage while retaining the discriminator in the "Conditional Generator training" stage?

In Sec 4.4.1, the paper says "This indicates that our method, which incorporates gradient backpropagation to the 3D representation, surpasses a pure 2D post-processing approach." Does the "gradient backpropagation to the 3D representation" only happen in the  "NeRF Optimization with Discriminator" stage as in the "Conditional Generator training" stage the NeRF gradient is detached?

In my current understanding of this sentence in the paper "However, our full method backpropagates to the 3D representation, thus reducing the added inconsistencies by over half.", I think the reason why GANeRF can obtain better view consistency than Nerfacto + ControlNet is that the GANeRF's input to the refinement network is of higher quality than Nerfacto + ControlNet due to the "NeRF Optimization with Discriminator" stage. So the refinement network in GANeRF actually learns to add fewer view-inconsistent details than Nerfacto + ControlNet, which makes GANeRF perform better. Am I right?

Thanks in advance.

**Comment by barbararoessle** [comment]
It is true, that gradients during generator training (stage 2) do not backpropagate to the NeRF, however, they do during the 1st training stage, where a discriminator provides feedback to the 3D representation. This already improves the results significantly over Nerfacto (see 'Ours w/o generator'). Hence GANeRF after the 1st training stage is a better basis for the 2D refinement than Nerfacto. Secondly, the GANeRF generator is well conditioned on the stage 1 results by using a multi-resolution conditioning. This helps to produce highly view-consistent results. 

Correct, 'Ours w/o discriminator' removes the discriminator in the 1st training stage, but keeps the generator training (with discriminator) unchanged.

Yes, gradient backpropagation to the 3D representation happens only in the 1st stage. 

Last paragraph is correct.

**Comment by 07hyx06** [comment]
Got it! Thanks.
